core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -124,8 +124,12 @@ class OptimizeExecutor(
 
       parallelJobCollection.tasksupport = forkJoinPoolTaskSupport
 
-      val updates = parallelJobCollection.flatMap(
-        partitionBinGroup => runCompactBinJob(txn, partitionBinGroup._1, partitionBinGroup._2)).seq
+      val updates = try {
```

Comment:
One small change. Store the reference to `ThreadUtils.newForkJoinPool(
        "OptimizeJob",
        sparkSession.sessionState.conf.getConf(DeltaSQLConf.DELTA_OPTIMIZE_MAX_THREADS))` and shutdown it directly.

Reply:
I actually just left for vacation so I won't be able to make this update. Feel free to close this and make the change yourself